corehq/motech/repeaters/repeater_generators.py

```diff
@@ -392,6 +394,176 @@ def submission_user_id(self):
         return CouchUser.get_by_username(self.submission_username()).user_id
 
 
+@attr.s
+class CaseUpdateConfig:
+    PROPS = {
+        "registry_slug": "target_data_registry",
+        "domain": "target_domain",
+        "case_type": "target_case_type",
+        "case_id": "target_case_id",
+        "includes": "target_property_includelist",
+        "excludes": "target_property_excludelist",
+        "override_props": "target_property_override",
+        "index_case_id": "target_index_case_id",
+        "index_case_type": "target_index_case_type",
+        "index_relationship": "target_index_relationship",
+    }
+
+    intent_case = attr.ib()
+    registry_slug = attr.ib()
+    domain = attr.ib()
+    case_type = attr.ib()
+    case_id = attr.ib()
+    includes = attr.ib()
+    excludes = attr.ib()
+    override_props = attr.ib()
+    index_case_id = attr.ib()
+    index_case_type = attr.ib()
+    index_relationship = attr.ib()
+
+    @classmethod
+    def from_payload(cls, payload_doc):
+        kwargs = {
+            attr: payload_doc.get_case_property(prop_name)
+            for attr, prop_name in cls.PROPS.items()
+        }
+        kwargs["intent_case"] = payload_doc
+        config = CaseUpdateConfig(**kwargs)
+        config.validate()
+        return config
+
+    def validate(self):
+        if self.includes is not None and self.excludes is not None:
+            raise DataRegistryCaseUpdateError("Both exclude and include lists specified. Only one is allowed.")
+        if self.includes is None and self.excludes is None:
+            raise DataRegistryCaseUpdateError("Neither exclude and include lists specified. One is required.")
+
+    def get_case_block(self, target_case):
+        return CaseBlock(
+            create=False,
+            case_id=target_case.case_id,
+            update=self.get_case_updates(target_case),
+            index=self.get_case_index(target_case)
+        ).as_text()
+
+    def get_case_updates(self, target_case):
+        case_json = self.intent_case.case_json
+        update_props = []
+        if self.excludes is not None:
+            update_props = set(case_json) - set(self.excludes.split())
+        if self.includes is not None:
+            update_props = set(case_json) & set(self.includes.split())
+
+        update_props.difference_update(set(self.PROPS.values()))
+        override = self.override_props is None or self.override_props.lower() in ("1", "true")
+        if not override:
+            target_props = set(target_case.case_json)
```

Comment:
Shouldn't this set exclude properties where the value is `''`?  I believe our standard thinking there is to treat blank properties the same as missing properties.